src/hooks/use-change-volume.ts

```diff
@@ -0,0 +1,73 @@
+import { useAndRequireContext } from "~/hooks/use-and-require-context.js";
+import { useEffect } from "react";
+import { ContextRef } from "~/containers/sound-provider/index.js";
+import { DEFAULT_VOLUME } from "~/constants/index.js";
+
+export const useChangeMasterVolume = () => {
+  const { audioContext, masterGainNode, masterVolume, setMasterVolume } =
+    useAndRequireContext(ContextRef, "useChangeMasterVolume", "sound-provider");
+
+  useEffect(() => {
+    if (!audioContext || !masterGainNode) {
+      return;
+    }
+    const storedMasterVolume = localStorage.getItem("MASTER_VOLUME");
+    const value =
+      storedMasterVolume !== null ? Number(storedMasterVolume) : DEFAULT_VOLUME;
+    masterGainNode.gain.setValueAtTime(Number(value), 0);
+    masterGainNode.connect(audioContext.destination);
+  }, [audioContext, masterGainNode, masterVolume]);
+
+  const changeMasterVolume = (volume: number) => {
+    localStorage.setItem("MASTER_VOLUME", String(volume));
```

Comment:
What if this gets executed on server side where there is no `localStorage`?

Reply:
Added isClient condition